ios/browser/api/ai_chat/ai_chat.mm

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/api/ai_chat/ai_chat.h"
+
+#include "ai_chat.mojom.objc+private.h"
+#include "base/functional/bind.h"
+#include "base/memory/raw_ptr.h"
+#include "base/strings/sys_string_conversions.h"
+#include "brave/base/mac/conversions.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_feedback_api.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_metrics.h"
+#include "brave/components/ai_chat/core/browser/constants.h"
+#include "brave/components/ai_chat/core/browser/models.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-shared.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "brave/ios/browser/api/ai_chat/ai_chat_conversation_driver.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "ios/chrome/browser/shared/model/application_context/application_context.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/chrome/common/channel_info.h"
+#include "services/data_decoder/public/cpp/data_decoder.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+@interface AIChat () {
+  raw_ptr<ChromeBrowserState> browser_state_;
+  std::unique_ptr<ai_chat::AIChatMetrics> ai_chat_metrics_;
+  std::unique_ptr<ai_chat::AIChatFeedbackAPI> ai_chat_feedback_api_;
+  std::unique_ptr<ai_chat::AIChatConversationDriverIOS> driver_;
+}
+@end
+
+@implementation AIChat
+- (instancetype)initWithChromeBrowserState:(ChromeBrowserState*)browserState
+                                  delegate:
+                                      (__weak id<AIChatDelegateIOS>)delegate {
+  if ((self = [super init])) {
+    browser_state_ = browserState;
+    ai_chat_metrics_ = std::make_unique<ai_chat::AIChatMetrics>(
+        GetApplicationContext()->GetLocalState());
+    ai_chat_feedback_api_ = std::make_unique<ai_chat::AIChatFeedbackAPI>(
+        browserState->GetSharedURLLoaderFactory(), GetChannelString());
+    driver_ = std::make_unique<ai_chat::AIChatConversationDriverIOS>(
+        delegate, browserState, ai_chat_metrics_.get());
+  }
+  return self;
+}
+
+- (bool)isAgreementAccepted {
+  auto* pref_service = user_prefs::UserPrefs::Get(browser_state_);
```

Comment:
Probably better to just get the pref service and store it for the lifetime of the instance, no?